spark/v3.2/build.gradle

```diff
@@ -58,7 +58,9 @@ project(":iceberg-spark:iceberg-spark-${sparkMajorVersion}_${scalaVersion}") {
     implementation project(':iceberg-orc')
     implementation project(':iceberg-parquet')
     implementation project(':iceberg-arrow')
-    implementation("org.scala-lang.modules:scala-collection-compat_${scalaVersion}")
+    implementation("org.scala-lang.modules:scala-collection-compat_${scalaVersion}") {
```

Comment:
`scala-collection-compat` is included into spark binary release tarball, so `compileOnly` should be used here IMO

Reply:
I too thought of it. But spark/v3.2/spark-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ExtendedDistributionAndOrderingUtils.scala needs it at runtime to support scala 2.13 (from PR#4009). So, I have just excluded scala-library files.

Reply:
Then I think we need not touch this part, just exclude it from the runtime jar

Reply:
updated.